manager/orchestrator/service.go

```diff
@@ -44,8 +46,11 @@ func DeleteServiceTasks(ctx context.Context, s *store.MemoryStore, service *api.
 	err = s.Batch(func(batch *store.Batch) error {
 		for _, t := range tasks {
 			err := batch.Update(func(tx store.Tx) error {
-				if err := store.DeleteTask(tx, t.ID); err != nil {
-					log.G(ctx).WithError(err).Errorf("failed to delete task")
+				// update desired state to REMOVE
+				t.DesiredState = api.TaskStateRemove
```

Comment:
It's good practice to make sure that the state is being advanced, and never moves backwards. Currently there isn't a state beyond "remove" but we could potentially add one later.